### PR TITLE
Verify diagnostics in parsing tests

### PR DESCRIPTION
**Pull request by Youssef1313**
Fixes https://github.com/dotnet/roslyn/issues/61586

### CODE REVIEW

File: src/Compilers/CSharp/Test/Syntax/Parsing/ParsingTests.cs [hunk 1]
```diff
@@ -146,24 +141,18 @@ internal void UsingExpression(string text, params DiagnosticDescription[] expect
...
         }
 
-        /// <summary>
-        /// Parses given string and initializes a depth-first preorder enumerator.
-        /// </summary>
-        protected SyntaxTree UsingTree(string text, CSharpParseOptions? options = null)
+        protected SyntaxTree UsingTree(string text, params DiagnosticDescription[] expectedErrors)
         {
-            return UsingTree(text, options, verifyErrors: false, expectedErrors: null);
+            return UsingTree(text, null, expectedErrors);
```
Comment (RikkiGibson) [inline src/Compilers/CSharp/Test/Syntax/Parsing/ParsingTests.cs:146]: consider labeling the `null` argument here

Comment (Youssef1313) [inline src/Compilers/CSharp/Test/Syntax/Parsing/ParsingTests.cs:146]:
```suggestion
            return UsingTree(text, options: null, expectedErrors);
```

File: src/Compilers/CSharp/Test/Syntax/Parsing/ParsingTests.cs [hunk 2]
```diff
@@ -146,24 +141,18 @@ internal void UsingExpression(string text, params DiagnosticDescription[] expect
...
         {
             VerifyEnumeratorConsumed();
             var tree = ParseTree(text, options);
             _node = tree.GetCompilationUnitRoot();
-            if (verifyErrors)
-            {
-                var actualErrors = _node.GetDiagnostics();
-                actualErrors.Verify(expectedErrors);
-            }
+            var actualErrors = _node.GetDiagnostics();
```
Comment (RikkiGibson) [inline src/Compilers/CSharp/Test/Syntax/Parsing/ParsingTests.cs:154]: For reviewers: this is the important part of the change. It's no longer possible to specify 'verifyErrors: false'.

File: src/Compilers/CSharp/Test/Syntax/Parsing/SeparatedSyntaxListParsingTests.cs [hunk 3]
```diff
@@ -39,7 +39,25 @@ class C
     A<,T> a7;
     A<T U,,> a8;
 }
-");
+",
+                // (7,9): error CS1003: Syntax error, ',' expected
+                //     A<T U> a4;
+                Diagnostic(ErrorCode.ERR_SyntaxError, "U").WithArguments(",").WithLocation(7, 9),
```
Comment (RikkiGibson) [inline src/Compilers/CSharp/Test/Syntax/Parsing/SeparatedSyntaxListParsingTests.cs:45]: perhaps the original authors of the tests didn't care about the errors, but it feels like this is a good thing. It feels like if someone wants to in the future they could always introduce a 'UsingTreeIgnoreErrors' method.

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 4)

**Pull request merged**